feat: add readme (#1)
Eureka-Shoulders/componentz: README.md

# Componentz

![npm](https://img.shields.io/npm/v/@euk-labs/componentz)
![NPM](https://img.shields.io/npm/l/@euk-labs/componentz)
![GitHub Workflow Status](https://img.shields.io/github/workflow/status/Eureka-Shoulders/componentz/CI)
![npm](https://img.shields.io/npm/dw/@euk-labs/componentz)

Componentz is an MUI-based UI library that can provide simple and performative ways to use components.

The goal is not to replace the MUI, but to be an extension

## Installation

> MUI is the base of this library so it is the first installation

Reviewer: I guess it will be better if you include a link to MUI Docs and Installation Guide

```bash
# Using npm:
npm install @mui/material @emotion/react @emotion/styled

# Using yarn:
yarn add @mui/material @emotion/react @emotion/styled
```

> For dependency injection we will use the inversify lib

Reviewer: Also provide the Inversify Docs link for users know more about what are we doing

```bash
# Using npm:
npm i inversify inversify-react
# Using yarn:
yarn add inversify inversify-react
```

> For state management we will use MobX

Reviewer: Maybe it's too big and boring???
What about tell which libs are necessary and why we need before giving devs a big command with all they need?

```bash
# Using npm:
npm i mobx mobx-react-lite

# Using yarn:
yarn add mobx mobx-react-lite
```

> Finlay componentz

Reviewer: ```suggestion
Finally componentz
```

```bash
# Using npm:
npm i mobx mobx-react-lite

# Using yarn:
yarn add mobx mobx-react-lite
```

### Dependencies

- MUI: A UI library that serves as the basis for the project
- MobX is used to generate Stores and Hooks powered by these stores to make a beautiful reactive way to control the data on your interfaces.

> - Inversify: dependency Injection library

Reviewer: ```suggestion
- Inversify is fundamental for dependency injection strategies
```

## Usage

> We've separated the logical part of the component, which has the minimum responsibilities to be as generic as possible, so we use the Mobx state manager to make this component control.

Reviewer: ```suggestion
We've separated the logical part of the component, which has the minimum responsibilities to be as generic as possible, so we use Mobx for state management to make this happen.
```

> The complete documentation form components is being built, we will soon make it available.

Reviewer: ```suggestion
The complete documentation is under construction, we will soon make it available.
```

> First add a Inversify provider and reflect metadata at the root of your project

Reviewer: ```suggestion
First add a Inversify provider and reflect-metadata at the root of your project
```

```ts
import 'reflect-metadata';

import { ThemeProvider } from '@mui/material/styles';
import { CssBaseline } from '@mui/material';
import { Provider } from 'inversify-react';
import theme from 'styles/theme';

> function MyApp({ Component, pageProps }) {

Reviewer: Specify that this is a NextJS example

  return (

>     <Provider container={globalContainer}>

Reviewer: Where `globalContainer` is defined?

![GitHub Workflow Status](https://img.shields.io/github/workflow/status/Eureka-Shoulders/componentz/CI)
![npm](https://img.shields.io/npm/dw/@euk-labs/componentz)

Componentz is a MUI-based UI library that can provide simple and performatic ways to use components.

The goal is not to replace the MUI, but to be an extension

### Dependencies

- MUI: A UI library that serves as the basis for the project
  - You can learn more about MUI [here](mui.com/)
- MobX is used to generate Stores and Hooks powered by these stores to make a beautiful reactive way to control the data on your interfaces.

>   - Meet this wonderfully lib [here](mobx.js.org/)

Reviewer: ```suggestion
  - Meet this wonderful lib [here](mobx.js.org/)
```

- Inversify is fundamental for dependency injection strategies

>   - The strategy about this architecture can be known [here](https://github.com/inversify/InversifyJS)

Reviewer: ```suggestion
  - The strategy about this architecture can be found [here](https://github.com/inversify/InversifyJS)
```
